cloud/pkg/cloudhub/servers/httpserver/upgrade.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The KubeEdge Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package httpserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/emicklei/go-restful"
+	"k8s.io/klog/v2"
+
+	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
+	beehiveModel "github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/cloud/pkg/common/modules"
+	commontypes "github.com/kubeedge/kubeedge/common/types"
+)
+
+// upgradeEdge upgrade the edgecore version
+func upgradeEdge(request *restful.Request, response *restful.Response) {
+	resp := commontypes.NodeUpgradeResponse{}
+
+	defer func() {
+		if _, err := response.Write([]byte("ok")); err != nil {
+			klog.Errorf("failed to send upgrade edge resp, err: %v", err)
+		}
+	}()
+
+	body, err := io.ReadAll(request.Request.Body)
```

Comment:
io.ReadAll will cause a memory exhaustion problem.
Please limit the read content by io.LimitedReader, refer to https://github.com/kubeedge/kubeedge/blob/master/edge/pkg/metamanager/metaserver/handlerfactory/handler.go#L211